lang/java/reef-utils/src/test/java/org/apache/reef/util/MultiAsyncToSyncTest.java

```
@@ -0,0 +1,240 @@
+
+    SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+    final Integer result = apiObject.apiCall(input);
+    apiObject.complete();
+
+    Assert.assertTrue(result.equals(0));
+  }
+
+  /**
+   * Used to make multiple asynchronous calls to the synchronous API.
+   */
+  final class AsynchronousCaller implements Callable<Boolean> {
+    private final SynchronousApi apiObject;
+    private final Integer input;
+
+    /**
+     * @param apiObject The synchronous api to be calle.
+     * @param input Input value for the api call.
+     */
+    AsynchronousCaller(final SynchronousApi apiObject, final Integer input) {
+      this.apiObject = apiObject;
+      this.input = input;
+    }
+
+    /**
+     * @return A Boolean object that indicates whether
+     *         or not the processing was successful.
+     * @throws Exception
+     */
+    public Boolean call() throws Exception {
+      final Integer result = apiObject.apiCall(input);
+      return result.equals(input + 1);
+    }
+  }
+
+  /**
+   * Verify no interaction occurs when multiple calls are in flight.
+   */
+  @Test
+  public void testMulitpleCalls() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 2;
+    final long timeoutPeriodSeconds = 4;
+    final Integer input = 1;
+    final ExecutorService executor = Executors.newFixedThreadPool(2);
+
+    SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+    FutureTask<Boolean> task1 = new FutureTask<>(new AsynchronousCaller(apiObject, input));
+    FutureTask<Boolean> task2 = new FutureTask<>(new AsynchronousCaller(apiObject, input + 1));
+
+    // Execute API calls concurrently.
+    executor.execute(task1);
+    executor.execute(task2);
+
+    try {
+      Assert.assertTrue(task1.get());
+      Assert.assertTrue(task2.get());
+    } catch (Exception e) {
```

let it be thrown out of the test method